src/util/string_utils.cpp

```diff
@@ -162,17 +162,21 @@ std::string escape(const std::string &s)
 
 /// Replace all occurrences of a string inside a string
 /// \param [out] str: string to search
-/// \param from: string to replace
+/// \param from: string to replace; must be non-empty
 /// \param to: string to replace with
 /// Copyright notice:
 /// Attributed to Gauthier Boaglio
 /// Source: https://stackoverflow.com/a/24315631/7501486
 /// Used under MIT license
+/// Note that this is quadratic in str.size() if from.size() != to.size().
```

Comment:
... , because each replacement entails a copy of the contents of the string.

Or maybe something else?

Reply:
No, that would not change the complexity in str.size() -- unfortunately, std::string::replace does not provide the guarantee that replacing equal-length strings is not independent of the length of the containing string.